retriever/__main__.py

```diff
@@ -180,26 +181,29 @@ def main():
             sys.tracebacklimit = 0
 
         if hasattr(args, 'debug') and args.not_cached:
-            engine.use_cache = False
+            use_cache = False
         else:
-            engine.use_cache = True
-
+            use_cache = True
+        engine.use_cache = use_cache
         if args.dataset is not None:
             scripts = name_matches(script_list, args.dataset)
         else:
             raise Exception("no dataset specified.")
         if scripts:
-            for dataset in scripts:
-                print("=> Installing", dataset.name)
-                try:
-                    dataset.download(engine, debug=debug)
-                    dataset.engine.final_cleanup()
-                except KeyboardInterrupt:
-                    pass
-                except Exception as e:
-                    print(e)
-                    if debug:
-                        raise
+            if args.dataset.endswith('.zip'):
```

Comment:
@apoorvaeternity, I guess we are testing if a file is a zip, but would this be the best or optimal way to do that. What if I create a text file as `textile.zip`, what would happen?

Reply:
It will raise `zipfile.BadZipFile: File is not a zip file` error. I think we can print a better message.

Reply:
Also something I noted is that we are raising, this will stop the whole process especially when we interface with other languages like R or Julia. I think we could make good use of `return` in this case.

> It will raise zipfile.BadZipFile: File is not a zip file error. I think we can print a better message.

I will test `textile.zip`, and let you know what is happening.

Reply:
Yes. I will make the change so that we do not raise the errors.